File: C#/0459. Repeated Substring Pattern.cs
public class Solution {
    public bool RepeatedSubstringPattern(string s) {
        bool flag=false;
        for(int i=1;i<s.Length;i++){
            if(s.Length%i==0 && IsRepeatedSubtring(s,s.Substring(0,i),i)){
                flag=true;
                break;
            }
        }
        return flag;
    }
    
    public bool IsRepeatedSubtring(string s,string str,int k){
        if(s.Length==0){
            return true;
        }
        if(s.Substring(0,k)!=str){
            return false;
        }
        return IsRepeatedSubtring(s.Substring(k),str,k);
    }
}

File: 20190519python/0459. Repeated Substring Pattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        if len(s)<=1:
            return False
        for k in range(1,len(s)//2+1):
            if len(s)%k==0 and s[:k]*(len(s)//k)==s:
                return True
        return False
    
